grafana-plugin/src/containers/RotationForm/RotationForm.tsx

```diff
@@ -477,7 +479,9 @@ export const RotationForm = observer((props: RotationFormProps) => {
 
       const isMonthlyRecurrence = shift.frequency === RepeatEveryPeriod.MONTHS;
       const activeOnSelectedPartOfDay =
-        repeatEveryInSeconds(shift.frequency, shift.interval) !== shiftEnd.diff(shiftStart, 'seconds') &&
```

Comment:
This check: 
```js
repeatEveryInSeconds(shift.frequency, shift.interval) !== shiftEnd.diff(shiftStart, 'seconds'))
``` 
only works when `Mask by weekdays` is disabled. If `Mask by weekdays` is enabled, we should activate the `Limit each shift length` switch only when the limit is less than 24 hours.